Applications/Spire/Include/Spire/Ui/RecentColors.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef SPIRE_RECENT_COLORS_HPP
+#define SPIRE_RECENT_COLORS_HPP
+#include <array>
+#include <deque>
+#include <QColor>
+
+namespace Spire {
+
+  //! Represents a data structure for storing recently selected colors in a
+  //  FIFO manner.
+  class RecentColors {
```

Comment:
Add default move constructor and assignment. Then, you can use it for copy-and-move everywhere an object of `RecentColors` is acquired.